drivers/sensors/cirque_pinnacle_gestures.c

```diff
@@ -204,9 +218,216 @@ void cirque_pinnacle_configure_circular_scroll(uint8_t outer_ring_pct, uint8_t t
 }
 #endif
 
+#if defined(CIRQUE_PINNACLE_DPAD_MODE_ENABLE)
+static dpad_state_t dpad = {
+    .keystate = 0,
+    .discrete = true,
+};
+void dpad_emit_key(dpad_key_t key, bool press) {
```

Comment:
A few issues here,
* The coupling of drivers to keymap and/or action code is conceptually wrong and would not be accepted.
* The implementation as it stands will result in keyboards where layout macros are padded with dummy locations (eg the mess before pre encoder-map was implemented).

Reply:
okay - so i'll need to look into how the encoders work, and rework the implementation to do sth similar...

thnx for the input; expect a v2 :-)

Reply:
Its potentially going to look a lot like #21426 (though i think that PR could do with some changes).

Reply:
Yeah, this kind of felt like a more hardware specific version of that.   The more general approach would be better, as more gain from it, not just the cirque.

Reply:
> Its potentially going to look a lot like #21426 
thanks for the hint! -> looking at/trying out that PR

> (though i think that PR could do with some changes).
do tell (there) what you have in mind (-: